backend/api/communication/src/main/java/co/airy/core/api/communication/SendMessageController.java

```diff
@@ -105,4 +108,26 @@
 
         return ResponseEntity.status(s).body(MessageResponsePayload.fromMessageContainer(new MessageContainer(message, new MetadataMap())));
     }
+
+    @PostMapping("/messages.resend")
+    public ResponseEntity<?> sendMessage(@RequestBody @Valid ResendMessageRequestPayload payload) {
+        final String messageId = payload.getMessageId().toString();
+
+        final MessageContainer messageContainer = stores.getMessageContainer(messageId);
+        if (messageContainer == null) {
+            return ResponseEntity.status(HttpStatus.NOT_FOUND).build();
+        }
+        if (!messageContainer.getMessage().getDeliveryState().equals(DeliveryState.FAILED)) {
+            return ResponseEntity.status(HttpStatus.CONFLICT).body(new RequestErrorResponsePayload("Message is not in a failed state"));
+        }
+
+        final Message message = MessageRepository.markMessageForResend(messageContainer.getMessage());
+        try {
+            stores.storeMessage(message);
+        } catch (InterruptedException | ExecutionException e) {
+            return ResponseEntity.status(HttpStatus.INTERNAL_SERVER_ERROR).body(e.getMessage());
+        }
+
+        return ResponseEntity.status(HttpStatus.ACCEPTED).build();
```

Comment:
I think it will be better to return `200`. I know that `stores.storeMessage(message)` is async. but to keep consistency with other endpoint in this app and the rest